docs/guides/staking/overview.md

```diff
@@ -0,0 +1,213 @@
+i.e., share-to-atom exchange rate can be set to any meaningful value), i.e., 
+share-to-atom-ex-rate = 1 atom per share. Then at the global pool level we 
+have, the size of the pool is 40 Atoms, and the amount of issued shares is 
+equal to 40. And for each validator we store in their corresponding data 
+structure that each has 10 shares of the bonded pool. Now lets assume that the 
+validator p4 starts process of unbonding of 5 shares. Then the total size of 
+the pool is decreased and now it will be 35 shares and the amount of Atoms is 
+35 . Note that the only change in other data structures needed is reducing the 
+number of shares for a validator p4 from 10 to 5.
+
+Let's consider now the case where a validator p1 wants to bond 15 more atoms to
+the pool. Now the size of the pool is 50, and as the exchange rate hasn't 
+changed (1 share is still worth 1 Atom), we need to create more shares, i.e. we
+now have 50 shares in the pool in total. Validators p2, p3 and p4 still have 
+(correspondingly) 10, 10 and 5 shares each worth of 1 atom per share, so we 
+don't need to modify anything in their corresponding data structures. But p1 
+now has 25 shares, so we update the amount of shares owned by p1 in its 
+data structure. Note that apart from the size of the pool that is in Atoms, all
+other data structures refer only to shares.
+
+Finally, let's consider what happens when new Atoms are created and added to 
+the pool due to inflation. Let's assume that the inflation rate is 10 percent 
+and that it is applied to the current state of the pool. This means that 5 
+Atoms are created and added to the pool and that each validator now 
+proportionally increase it's Atom count. Let's analyse how this change is 
+reflected in the data structures. First, the size of the pool is increased and 
+is now 55 atoms. As a share of each validator in the pool hasn't changed, this 
+means that the total number of shares stay the same (50) and that the amount of
+shares of each validator stays the same (correspondingly 25, 10, 10, 5). But 
+the exchange rate has changed and each share is now worth 55/50 Atoms per 
+share, so each validator has effectively increased amount of Atoms it has.  So 
+validators now have (correspondingly) 55/2, 55/5, 55/5 and 55/10 Atoms. 
+
+The concepts of the pool and its shares is at the core of the accounting in the 
+Staking module. It is used for managing the global pools (such as bonding and 
+unbonding pool), but also for distribution of Atoms between validator and its 
+delegators (we will explain this in section X).
+
+#### Delegator shares
+
+A validator is, depending on its status, contributing Atoms to either the bond,
+unbonding or unbonded pool - the validator in turn holds some amount of pool
+shares.  Not all of a validators Atoms (and respective shares) are owned by the
+validator, some may be owned by delegators to that validator. The mechanism for
```

Comment:
"are *necessarily* owned", "some *or most* may be..."

Reply:
addressed